polxml

```diff
@@ -65,52 +65,128 @@ def intfromhexiflarge(a):
                 line = f.readline()
-            while 'ATOM' in line or 'HETATM' in line:
-                atom = {}
-                atom['n'] = intfromhexiflarge(line[6:11])
-                atom['name'] = line[12:16].strip()
-                atom['mol'] = line[17:20].strip()
-                atom['seq'] = intfromhexiflarge(line[22:26])
-                atom['x'] = float(line[30:38])
-                atom['y'] = float(line[38:46])
-                atom['z'] = float(line[46:54])
-                atom['ele'] = line[76:78].strip()
-                atom['nH'] = 0         # number of bonded H
-                atom['type'] = ''      # non-bonded type
-                atom['class'] = ''     # bonded type
-                self.atoms.append(atom)
+                while 'CRYST1' not in line:
+                    line = f.readline()
+                self.box = line.strip()
                 line = f.readline()
-            while 'CONECT' in line:
-                i = intfromhexiflarge(line[6:11]) - 1
-                j = intfromhexiflarge(line[11:16]) - 1
-                if self.atoms[i]['seq'] != self.atoms[j]['seq']:
-                    print(f'bond between atoms {i:d} and {j:d} of different molecules')
-                    exit(1)
-                self.bonds.append({'i': i, 'j': j})
+                while not (line.startswith('ATOM') or line.startswith('HETATM')):
+                    line = f.readline()
+                while 'ATOM' in line or 'HETATM' in line:
+                    atom = {}
+                    atom['n'] = intfromhexiflarge(line[6:11])
+                    atom['name'] = line[12:16].strip()
+                    atom['mol'] = line[17:20].strip()
+                    atom['seq'] = intfromhexiflarge(line[22:26])
+                    atom['x'] = float(line[30:38])
+                    atom['y'] = float(line[38:46])
+                    atom['z'] = float(line[46:54])
+                    atom['ele'] = line[76:78].strip()
+                    atom['nH'] = 0         # number of bonded H
+                    atom['type'] = ''      # non-bonded type
+                    atom['class'] = ''     # bonded type
+                    self.atoms.append(atom)
+                    line = f.readline()
+                while 'CONECT' in line:
+                    i = intfromhexiflarge(line[6:11]) - 1
+                    j = intfromhexiflarge(line[11:16]) - 1
+                    if self.atoms[i]['seq'] != self.atoms[j]['seq']:
+                        print(f'bond between atoms {i:d} and {j:d} of different molecules')
+                        exit(1)
+                    self.bonds.append({'i': i, 'j': j})
+                    line = f.readline()
+                self.natom = len(self.atoms)
+
+        if pdbfile == 'config.mmcif':
+            with open(pdbfile, 'r') as f:
                 line = f.readline()
-        self.natom = len(self.atoms)
-
-    def write(self, pdbfile):
+                for line in f:
```

Comment:
Again, we should not presume the order or de content of the columns: parse the lines after loop_ to find the content of the fields.